docs/designs/link-output.md

```diff
@@ -0,0 +1,97 @@
+
+## Links
+
+`links` includes all the links which has been processed by docfx:
+
+- Relative link in source content will be processed by docfx and convert it to absolute link if it's valid, otherwise keep it untouched.
+
+- Absolute link in source content will NOT processed by docfx
+
+| Links in source content | Examples              | Output in `links.json` |
+| ----------------------- | --------------------- | ---------------------- |
+| Relative Link           | ./a.md (valid)        | /basepath/article/a    |
+|                         | a/b.md (invalid)      | a/b.md                 |
+| Absolute Link           | /basepath/articles/b  | /basepath/articles/b   |
+|                         | www.github.com/x      | www.github.com/x       |
+|                         | https://docs.com/y    | https://docs/com/y     |
+
+> NOTE: the absolute links converted from relative links by docfx don't contains `locale` and `versioning` information
+
+##  Bookmarks
+
+`bookmarks` includes all bookmarks ids generated/defined from source content, includes its inclusions(token/code-snippets)
+
+| Source content examples        | Html content                             | Output in `links.json` |
+| ------------------------------ | -----------------------------------------| ---------------------- |
+|  `# Title H1`                  | `<h1 id="title-h1">Title H1</h1>`        | title-h1               |
+|  `## Title H2`                 | `<h2 id="title-h2">Title H2</h2>`        | title-h2               |
+|  `## Title H2`                 | `<h2 id="title-h2">Title H2</h2>`        | title-h2               |
+| `<a id="##special-chars"></a>` | `<a id="##special-chars"></a>`           | ##special-chars        |
+| `<a id="with:colon"></a>`      | `<a id="with:colon"></a>`                | with:colon             |
+| `[!include[](d.md)]`           | `<h1 id="title-h1">Title H1</h1>`        | title-h1               |
+
+> The content in d.md is `# Title H1` in above case
+
+## Output
+
+`links.json` file will be generated by docfx to contains all the links and bookmarks information about all the page files in build.
+
+The output is **an array** of below data model
+
+  |            |                      |
+  |----------- |--------------------- |
+  | `path`     | dotnet/articles/a.md |
+  | `site-url` | /dotnet/articles/a   |
+  | `monikers` | ['dotnet-1.1']       |
+  | `links`    | *link model*         |
+  | `bookmarks`| *bookmark model*     |
+
+*link model*
+
+  |            |                      |
+  |----------- |--------------------- |
+  | `link`     | dotnet/articles/b |
+
+*bookmark model*
+
+  |            |                      |
+  |----------- |--------------------- |
+  | `id`       | title-h1             |
+
+Example:
+
+```json
+[
+    {
+        "path": "dotnet/articles/a.md",
+        "site-url": "/dotnet/articles/a",
```

Comment:
Consider use `url` here, the meaning it the same as the `url` in `.publish.json`
```suggestion
        "url": "/dotnet/articles/a",
```